engine/src/main/java/com/datatorrent/stram/webapp/TypeGraph.java

```diff
@@ -355,7 +356,8 @@ public int size()
     }
     Set<String> result = new TreeSet<>();
     for (TypeGraphVertex node : tgv.allInstantiableDescendants) {
-      if ((isAncestor(InputOperator.class.getName(), node.typeName) || !getAllInputPorts(node).isEmpty())) {
+      if ((isAncestor(InputOperator.class.getName(), node.typeName) || isAncestor(Module.class.getName(), node.typeName)
```

Comment:
Method should do what the name suggest . Should either change the name of the method or have this in a separate method. There are same problem in TypeGraph and OperatorDiscover, please fix them as well. Otherwise, it would be confusing and error-prone in the future.